### ISSUE TITLE
Operator Hub outputs can be more user friendly than what they are right now

### ISSUE THREAD

**Issue by dharmit** [redhat-developer/odo#3197]
After looking at [odo & Operator Hub integration documentation](https://odo.dev/docs/operator-hub/), @jorgemoralespou pointed out that outputs for Operator Hub integration are not really user friendly. For example, why do we use `--crd` flag to start a service? CRD is a k8s concept that we should ideally abstract that for users since that's the goal of odo.

Based on the discussion, IMO, we could change it to `--service`. But there's a scope for further improvement. We should change the output starting from `odo catalog list services` and instead of below output:
```sh
odo catalog list services                                       
Operators available in the cluster
NAME                                    CRDs
etcdoperator.v0.9.4                     EtcdCluster, EtcdBackup, EtcdRestore
mongodb-enterprise.v1.5.2               MongoDB, MongoDBUser, MongoDBOpsManager
service-binding-operator.v0.1.1-282     ServiceBindingRequest
```
we should consider presenting below to the user:
```sh
odo catalog list services                                       
OPERATOR NAME                           SERVICES PROVIDED
etcdoperator.v0.9.4                     EtcdCluster, EtcdBackup, EtcdRestore
mongodb-enterprise.v1.5.2               MongoDB, MongoDBUser, MongoDBOpsManager
service-binding-operator.v0.1.1-282     ServiceBindingRequest
```

/kind code-refactoring
/area service-operators
/area component

**Comment by jorgemoralespou** [comment]
I would not name it "Operator Name" And "Services Provided" but something more along the lines of:
`provider` and `services`, and maybe remove the version out, as that's also implementation details. In an ideal world, services could still come from the service catalog. Maybe no longer in OpenShift, but still in k8s world. 

```
odo catalog list services                                       
PROVIDER                         SERVICES
etcdoperator                     EtcdCluster, EtcdBackup, EtcdRestore
mongodb-enterprise               MongoDB, MongoDBUser, MongoDBOpsManager
service-binding-operator         ServiceBindingRequest
```

Maybe worth considering not using the CSV name but rather the api-group (e.g. etcd.database.coreos.com)

**Comment by dharmit** [comment]
This is still valid and important. Should be addressed through #4242 but needs to be kept open.

/remove-lifecycle rotten
/lifecycle frozen

**Comment by kadel** [comment]
`odo catalog` was removed from v3.

tagging #5523 in case we decide to implement again in future

/close
